src/grpc/stream_infer_handler.cc

```diff
@@ -262,6 +262,12 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
       err = TRITONSERVER_InferenceRequestNew(
           &irequest, tritonserver_.get(), request.model_name().c_str(),
           requested_model_version);
+      state->inference_request_ = {
```

Comment:
do we need to check err here? Or is it guranteed irequest is set to null or not changed? (meaning is it safe to delete it with InferenceRequestDelete() if InferenceRequestNew() doesn't succeed?)

Reply:
The current TRITONSERVER_InferenceRequestNew implementation does not change the request pointer on error.
However, let's play it safe and place it within the check.